Automattic/wp-calypso
# Emails: Support in checkout for yearly mailboxes in Professional Email

#### Changes proposed in this Pull Request

Adds support to show correct information when buying annually billed subscription for a Professional Email mailbox.

#### Testing instructions

1. Go to a site with domains without email subscription
2. Click on 'Add email' for the desired site
3. Once you are in the comparison page, add at the end the following flag: `?flags=emails/new-email-comparison`
5. Add annual subscription for Professional Email
6. Check that the checkout contains the information highlighted in the screenshot below

![image](https://user-images.githubusercontent.com/5689927/147357386-b07646e0-ec7b-42d8-ac8d-d1d166ab8d22.png)

Also check that the Thank you Page shown at the end of the process is the following one:
![image](https://user-images.githubusercontent.com/5689927/147357740-f71b3074-c7cb-4e75-9e8a-8f0bb380ea57.png)

**Remarks**: it is only visible through the path /email/{domain}/purchase/{site}?flags=emails/new-email-comparison

Related to {1200182182542585-as-1201559712017966}
